src/Plugin/search_api/processor/StartMonth.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\openy_activity_finder\Plugin\search_api\processor;
+
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\search_api\Datasource\DatasourceInterface;
+use Drupal\search_api\Item\ItemInterface;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Drupal\search_api\Processor\ProcessorProperty;
+
+/**
+ * Adds the start month to the indexed data.
+ *
+ * @SearchApiProcessor(
+ *   id = "openy_af_start_month",
+ *   label = @Translation("Start Month"),
+ *   description = @Translation("Translates datetime values of session to an index of start month date"),
+ *   stages = {
+ *     "add_properties" = 0,
+ *   },
+ *   locked = false,
+ *   hidden = false,
+ * )
+ */
+class StartMonth extends ProcessorPluginBase {
+
+  const PROPERTY_NAME = 'search_api_af_start_month';
+
+  const BASE_DATE = '1970-01-01T';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    if (!$datasource) {
+      $definition = [
+        'label' => $this->t('Start Month'),
+        'description' => $this->t("Translates datetime values of session to an index of start month date"),
+        'type' => 'string',
+        'processor_id' => $this->getPluginId(),
+        'is_list' => FALSE,
+      ];
+      $properties[self::PROPERTY_NAME] = new ProcessorProperty($definition);
+    }
+
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    $object = $item->getOriginalObject();
+    $entity = $object->getValue();
+
+    if (!$entity->hasField('field_session_time')) {
+      return;
+    }
+
+    $paragraphs = $entity->field_session_time ? $entity->field_session_time->referencedEntities() : [];
```

Comment:
**Optional:**
Could be shorter:     `$paragraphs = $entity->field_session_time?->referencedEntities() ?? [];`